### PR TITLE
Address pkg_resources deprecation warnings

### PR DESCRIPTION
**Pull request by victorlin**
### Description of proposed changes

Previously, some warnings were visible upon running pytest.

### Related issue(s)

N/A, I came across this while looking at CI logs.

### Testing

- [x] Checks pass
- [x] Warnings no longer appear in CI logs (all but `devel/release` are covered by tests)
- [ ] The next release is successful

### Checklist

- [x] ~Add a message in [CHANGES.md](https://github.com/nextstrain/augur/blob/HEAD/CHANGES.md) summarizing the changes in this PR that are end user focused. Keep headers and formatting consistent with the rest of the file.~ N/A, no functional changes

### CODE REVIEW

**Comment by tsibley** [comment]
_(from [my comment](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1688577007271919?thread_ts=1688000418.282969&cid=C01LCTT7JNN) in Slack)_

The package resources stuff is a mess currently in Python. There's been a lot of churn.

One downside to the solution here is that the file is completely loaded into memory at least three times (pkgutil, StringIO, our eventual parsing of it). Using the newest `importlib.resources` API (and `importlib_resources` backport) is not much code and can avoid the downside above. Nextstrain CLI [does that](https://github.com/nextstrain/cli/commit/63c73443288a2c72bfdae4fb72625683509d6416).

File: augur/data/__init__.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+"""
+Resource files.
+"""
+# Copied without modification from nextstrain/cli/resources/__init__.py in the
+# Nextstrain CLI project¹.
+#
+# ¹ <https://github.com/nextstrain/cli/blob/a5dda9c0579ece7acbd8e2c32a4bbe95df7c0bce/nextstrain/cli/resources/__init__.py>
```
Comment (victorlin) [inline augur/data/__init__.py:7]: @tsibley thanks for the pointer here. I like your approach in Nextstrain CLI and decided to copy it over here with some small modifications.

I've noticed in the past you've copied license files as well. Can you advise on the best approach here? Augur and Nextstrain CLI are under different licenses, so I'm thinking Nextstrain CLI's MIT license should be copied over.

Comment (tsibley) [reply augur/data/__init__.py:7]: Yep, the license should be copied as well. Basically all open source licenses require that the license comes along with any copying.

File: augur/data/__init__.py [hunk 2]
```diff
@@ -29,4 +29,7 @@
 
 
 def as_file(path: str) -> ContextManager[Path]:
-    return _as_file(_files(__name__) / path)
+    return _as_file(files(path))
+
+def files(path: str):
+    return _files(__name__) / path
```
Comment (tsibley) [inline augur/data/__init__.py:35]: This commit should also update the header at the top of the file which says "Copied without modification" even though the commit modifies it. (However, modification is not strictly necessary if you'd rather avoid it… see below.)

File: augur/validate.py [hunk 3]
```diff
@@ -31,7 +30,7 @@ def load_json_schema(path):
     (located in augur/data)
     '''
     try:
-        schema = json.loads(resource_string(__package__, os.path.join("data", path)))
+        schema = json.loads(files(path).read_text(encoding="utf-8"))
```
Comment (tsibley) [inline augur/validate.py:33]: _non-blocking, just a note_

It doesn't matter for memory usage/read speed here, but we could use `augur.data.as_files()` as is here too:

```suggestion
        with as_file(path) as file, open(file, "r", encoding = "utf-8") as fh:
            schema = json.load(fh)
```

alternatively, keeping `augur.data.files()` but avoiding reading the whole file into memory:

```suggestion
        with files(path).open("r", encoding = "utf-8") as fh:
            schema = json.load(fh)
```

**Review by tsibley — CHANGES_REQUESTED** [review]

**Review by corneliusroemer — COMMENTED** [review]
Looks great to address this, have been noticing the deprecation warnings as well

File: augur/data/__init__.py (hunk 1 above)
Comment (victorlin) [reply augur/data/__init__.py]: Copied the license over in cecd5e9063003ae95d6c014af9234f50396607d5.

File: augur/data/__init__.py (hunk 2 above)
Comment (victorlin) [reply augur/data/__init__.py]: Updated to not modify, but also changed the wording to be more future-proof:

> Originally copied from nextstrain/cli/resources/__init__.py in the Nextstrain CLI project¹.

File: augur/validate.py (hunk 3 above)
Comment (victorlin) [reply augur/validate.py]: Thanks, I went with the first suggestion since `files` is a bit confusing to me.

**Review by tsibley — APPROVED** [review]

**Pull request merged**